### PR TITLE
Added deep linking capabilities using the webar:// scheme.

### PR DESCRIPTION
**Pull request by judax**
Change-Id: Ibd039907811491e8583c71435b1515e32c861586

### CODE REVIEW

File: android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java [hunk 1]
```diff
@@ -767,7 +756,28 @@ public void onProgressChanged(int progress)
...
-        return intent != null ? intent.getDataString() : null;
+        String urlString = null;
+        if (intent != null) {
+            // If there is an intent, retrieve data sent over with the intent
+            Uri data = intent.getData();
+            urlString = intent.getDataString();
+            // If there is a url and the scheme was "webar", that means that
+            // a url is passed to be loaded.
+            if (data != null &&
+                    data.getScheme().
```
Comment (lincolnfrog) [inline android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java:767]: very large indent

File: android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java [hunk 2]
```diff
@@ -767,7 +756,28 @@ public void onProgressChanged(int progress)
...
+                    toLowerCase(Locale.getDefault()).equals("webar")) {
+                // Remove the "webar://" sceheme from the url
+                urlString = urlString.substring(8);
+                // If the url lacks scheme, add http by default.
+                String urlStringLowerCase = urlString.toLowerCase(Locale.getDefault());
+                if (!urlStringLowerCase.contains("http://") &&
+                        !urlStringLowerCase.contains("https://") &&
```
Comment (lincolnfrog) [inline android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java:774]: very large indent

File: android_webview/test/shell/AndroidManifest.xml [hunk 3]
```diff
@@ -38,6 +38,18 @@
         <action android:name="android.intent.action.MAIN" />
         <category android:name="android.intent.category.LAUNCHER" />
       </intent-filter>
+      <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="http"/>
```
Comment (lincolnfrog) [inline android_webview/test/shell/AndroidManifest.xml:45]: test if it works with https

**Review by lincolnfrog — APPROVED** [review]

File: android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java (hunk 1 above)
Comment (judax) [reply android_webview/test/shell/src/org/chromium/android_webview/shell/AwShellActivity.java:767]: What do you mean? Any tips on how to resolve it?

**Pull request merged**